proto/types.proto

```diff
@@ -0,0 +1,140 @@
+syntax = "proto3";
+package lazyledger.proto;
+
+import "google/protobuf/timestamp.proto";
+
+message Block {
+  Header header = 1;
+  AvailableDataHeader availableDataHeader = 2;
+  AvailableData availableData = 3;
+  Commit lastCommit = 4;
+}
+
+message Header {
+  uint64 version = 1;
+  uint64 chain_id = 2;
+  uint64 height = 3;
+  google.protobuf.Timestamp time = 4;
+  BlockID last_block_id = 5;
+  HashDigest last_commit_root = 6;
+  HashDigest consensus_hash = 7;
+  HashDigest state_commitment = 8;
+  HashDigest available_data_root = 9;
+  Address proposer_address = 10;
+}
+
+message AvailableDataHeader {
+  repeated HashDigest available_data_commitments = 1;
+}
+
+message AvailableData {
+  TransactionData transaction_data = 1;
+  IntermediateStateRootData intermediate_state_root_data = 2;
+  EvidenceData evidence_data = 3;
+  MessageData message_data = 4;
+}
+
+message Commit {
+  uint64 height = 1;
+  uint64 round = 2;
+  BlockID block_id = 3;
+  repeated CommitSig signatures = 4;
+}
+
+message BlockID {
+  HashDigest header_root = 1;
+}
+
+message HashDigest {
+  bytes raw_data = 1;
+}
+
+message Address {
+  bytes raw_data = 1;
+}
+
+message CommitSig {
+  enum BlockIDFlag {
+    BLOCK_ID_FLAG_UNSPECIFIED = 0;
+    BLOCK_ID_FLAG_ABSENT = 1;
+    BLOCK_ID_FLAG_COMMIT = 2;
+    BLOCK_ID_FLAG_NIL = 3;
+  }
+  BlockIDFlag block_id_flag = 1;
+  Address validator_address = 2;
+  google.protobuf.Timestamp timestamp = 3;
+  Signature signature = 4;
+}
+
+message Signature {
+  bytes r = 1;
+  bytes vs = 2;
+}
+
+message TransactionData {
+  repeated WrappedTransaction wrapped_transactions = 1;
+}
+
+message WrappedTransaction {
+  uint64 index = 1;
+  Transaction transaction = 2;
+  uint64 message_start_index = 3;
+}
+
+message Transaction {
+  uint64 message_size = 1;
+  HashDigest message_share_roots_root = 2;
+}
```

Comment:
As far as I remembered, not all transactions need to refer to messages. But if they don't where does their data live?

Reply:
Transactions and evidence will be updated when those are fleshed out later.